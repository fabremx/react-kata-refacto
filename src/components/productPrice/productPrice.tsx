import { ReactElement } from "react";
import styles from "./productPrice.module.scss";

interface Props {
  price: number;
  priceAfterDiscount?: number;
  quantity: number;
}

export function ProductPrice({
  price,
  priceAfterDiscount,
  quantity,
}: Props): ReactElement {
  const getDiscountPrice = (
    price: number,
    priceAfterDiscount: number,
    quantity: number
  ) => (
    <>
      <span className={styles.stripped}>{`$ ${price * quantity}`}</span>
      {` $ ${priceAfterDiscount * quantity}`}
    </>
  );

  const getPrice = (price: number, quantity: number) => `$ ${price * quantity}`;

  return (
    <div data-testid="price">
      {priceAfterDiscount
        ? getDiscountPrice(price, priceAfterDiscount, quantity)
        : getPrice(price, quantity)}
    </div>
  );
}
