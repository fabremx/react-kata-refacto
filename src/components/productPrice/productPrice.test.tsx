import { render } from "@testing-library/react";
import { ProductPrice } from "./productPrice";

describe("ProductPrice", () => {
  it("should render correctly the component when price DOES NOT have discount", () => {
    // When
    const { asFragment } = render(<ProductPrice price={10} quantity={2} />);
    // Then
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render correctly the component when price have discount", () => {
    // When
    const { asFragment } = render(
      <ProductPrice price={10} priceAfterDiscount={7} quantity={2} />
    );
    // Then
    expect(asFragment()).toMatchSnapshot();
  });
});
