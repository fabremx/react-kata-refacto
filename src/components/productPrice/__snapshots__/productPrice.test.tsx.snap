// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductPrice should render correctly the component when price DOES NOT have discount 1`] = `
<DocumentFragment>
  <div
    data-testid="price"
  >
    $ 20
  </div>
</DocumentFragment>
`;

exports[`ProductPrice should render correctly the component when price have discount 1`] = `
<DocumentFragment>
  <div
    data-testid="price"
  >
    <span
      class="stripped"
    >
      $ 20
    </span>
     $ 14
  </div>
</DocumentFragment>
`;
